src/Transfer/Commons/Stream/Adapter/DirectoryStreamAdapter.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of Transfer.
+ *
+ * For the full copyright and license information, please view the LICENSE file located
+ * in the root directory.
+ */
+
+namespace Transfer\Commons\Stream\Adapter;
+
+use Transfer\Adapter\SourceAdapterInterface;
+use Transfer\Adapter\TargetAdapterInterface;
+use Transfer\Adapter\Transaction\Request;
+use Transfer\Adapter\Transaction\Response;
+
+class DirectoryStreamAdapter implements SourceAdapterInterface, TargetAdapterInterface
+{
+    /**
+     * @var string
+     */
+    private $path;
+
+    /**
+     * @param string $path
+     */
+    public function __construct($path)
+    {
+        $this->path = $path;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function receive(Request $request)
+    {
+        $rii = new \RecursiveIteratorIterator(new \RecursiveDirectoryIterator($this->path));
+        $files = array();
+        foreach ($rii as $file) {
+            if ($file->isDir()) {
+                continue;
+            }
+            $stream = fopen($file->getPathname(), 'r');
+            $files[] = stream_get_contents($stream);
```

Comment:
A problem with this approach is that all of the files are loaded within the receive call, which might take some time depending on how many files there are. The receive method should instead return an iterator which will do the work of loading files when the `current()` function is called. See here: https://github.com/transfer-framework/transfer/blob/1.0/src/Transfer/Adapter/LocalDirectoryAdapter.php#L21